packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -457,12 +481,31 @@ export class HttpApi extends HttpApiBase {
    */
   public addRoutes(options: AddRoutesOptions): HttpRoute[] {
     const methods = options.methods ?? [HttpMethod.ANY];
-    return methods.map((method) => new HttpRoute(this, `${method}${options.path}`, {
-      httpApi: this,
-      routeKey: HttpRouteKey.with(options.path, method),
-      integration: options.integration,
-      authorizer: options.authorizer,
-      authorizationScopes: options.authorizationScopes,
-    }));
+    return methods.map((method) => {
+      let authorizationScopes = this.authorizationScopes;
+
+      if (Array.isArray(options.authorizationScopes)) {
+        authorizationScopes = options.authorizationScopes;
+      }
+
+      return new HttpRoute(this, `${method}${options.path}`, {
+        httpApi: this,
+        routeKey: HttpRouteKey.with(options.path, method),
+        integration: options.integration,
+        authorizer: options.authorizer ?? this.authorizer,
+        authorizationScopes,
+      });
+    });
+  }
+}
+
+/**
+ * Used to remove the default authorizer for a route
+ */
+export class NoneAuthorizer implements IHttpRouteAuthorizer {
```

Comment:
Since this package is hosting both HTTP APIs and [(future) websocket APIs](https://github.com/aws/aws-cdk/pull/13031), we should call this `HttpNoneAuthorizer`.

It's more appropriate to put this into the `authorizer.ts` file. And add a simple unit test.

Reply:
Done. The unit test is in api https://github.com/aws/aws-cdk/pull/13172/files#diff-afbfe319d751967cb29a7ec7bcf2f7a3ba6642b8ad5d3c14f40c17704459ad09R419